packages/hyperion-autologging/src/ALInteractableDOMElement.ts

```diff
@@ -182,16 +182,83 @@ const extractInnerText = (element: HTMLElement | null): string | null => {
   return null;
 };
 
+export type ALElementNameResult = Readonly<{
+  text: string,
+  source: 'innerText' | 'aria-label' | 'aria-labelledby'| 'aria-description' | 'aria-describedby';
+}>;
+
 export function getElementName(
   element: HTMLElement,
   // Whether to try to resolve if element name is a result of Fbt translation
-): string | null {
+): ALElementNameResult | null {
   let nextElement: HTMLElement | null = element;
   while (nextElement && nextElement.nodeType === Node.ELEMENT_NODE) {
+
+    const labelledBy = nextElement.getAttribute('aria-labelledby');
+    if (labelledBy != null) {
+      const fullLabel = labelledBy.split(' ').map(labelledBy => {
+        const labelElement = document.getElementById(labelledBy);
+        if (labelElement != null) {
+          const labelText = extractInnerText(labelElement);
+          if (labelText != null) {
+            return {
+              text: labelText,
+            };
+          }
+        }
+        return null;
+      });
+      const fullLabelText = fullLabel.map(label => label?.text).join(' ');
```

Comment:
I think you also need to filter null, or better yet, just return '' (empty string) instead of null so that when you call `.join` it works well.